spec/acceptance/05_enterprise_spec.rb

```diff
@@ -28,6 +28,57 @@ class { 'sensu::backend':
       end
     end
 
+    describe command('sensuctl license info'), :node => node do
+      its(:exit_status) { should eq 0 }
+    end
+  end
+  context 'updates license file' do
+    it 'should work without errors and be idempotent' do
+      pp = <<-EOS
+      class { 'sensu::backend':
+        license_source => '/root/sensu_license.json',
+      }
+      EOS
+
+      # Remove license file to ensure refresh works
+      on node, puppet("resource file /etc/sensu/license.json ensure=absent")
+      if RSpec.configuration.sensu_use_agent
+        site_pp = "node 'sensu_backend' { #{pp} }"
+        puppetserver = hosts_as('puppetserver')[0]
+        create_remote_file(puppetserver, "/etc/puppetlabs/code/environments/production/manifests/site.pp", site_pp)
+        on node, puppet("agent -t --detailed-exitcodes"), acceptable_exit_codes: [2]
+      else
+        apply_manifest_on(node, pp, :expect_changes => true)
+      end
+    end
+
+    describe command('sensuctl license info'), :node => node do
+      its(:exit_status) { should eq 0 }
+    end
+  end
+  context 're-adds license file' do
+    it 'should work without errors and be idempotent' do
+      pp = <<-EOS
+      class { 'sensu::backend':
+        license_source => '/root/sensu_license.json',
+      }
+      EOS
+
+      # Remove license to verify it can re-add
+      on node, "sensuctl delete -f /etc/sensu/license.json"
```

Comment:
Use the new type's ensure => absent instead of calling the sensuctl delete directly, so we have a test for that functionality.

Reply:
The type supports ensure=absent but the type is only used through `sensu::backend` and that class doesn't support setting the license to absent. I updated the acceptance test to use `ensure=absent` but it's not intended to be used.  I didn't want to add logic to the backend class that might remove someone's license which could do nasty things if they have more than 100 nodes.